lix/packages/sdk/src/change-queue.test.ts

```diff
@@ -118,7 +82,20 @@ test("should use queue and settled correctly", async () => {
 	const changes = await lix.db
 		.selectFrom("change")
 		.innerJoin("snapshot", "snapshot.id", "change.snapshot_id")
-		.selectAll()
+		.select([
+			"change.id",
+			"change.author",
+			"change.created_at",
+			"change.snapshot_id",
+			"change.parent_id",
+			"change.entity_id",
+			"change.type",
+			"change.file_id",
+			"change.plugin_key",
+			"snapshot.value as value",
+			"change.meta",
+			"change.commit_id",
+		])
```

Comment:
This is required because an 		.innerJoin("snapshot", "snapshot.id", "change.snapshot_id") select all overshadows the change.id with the snapshot.id

Reply:
Oh god what a beast - this combined with flaky run envs in test (importing from dist...) Sounds like fun.

Reply:
I looked into the code - seems there are a couple more places where this is still a problem

Reply:
I wonder if we can use some bits in the uuid v7 to encode the entity type (like id of a change vs id of a snapshot) - In a former project we used some of the machine id bits in [snowflake](https://en.wikipedia.org/wiki/Snowflake_ID) to signal the type of the id. This allowed us to runtime assert correct entities early based on passed id's - saved us hours, if not days of debugging and post mortems. 

**Not the solution** - and I need into uuid v7 but to outline what I have in mind:

If every id of a change must end with a C
40e97286-9b78-4a35-8fb1-0036bdfbbf7C
Every id of a snapshot ends with an S
40e97286-9b78-4a35-8fb1-0036bdfbbf7S

We could assert in a function like getLeafChange that the id is an expected one:
```
getLeafChange(change) {
  if (!change.id.endsWith('C')) {
    throw new WrongObjectTypePassed("The Id passed is not the id of a change")
  } 
 ....
}

```

Reply:
I double checked kysely we can make use selectAll with 'change' as parameter and and an explicit select
```suggestion
+		.selectAll('change')
+		.select('snapshot.value')
```
which is translated to:

```
SELECT
  "change".*,
  "snapshot"."value"
```

I will commit on this pr and fix the other places where we make use of innerjoin as well

Reply:
Oh wow I was looking for `select('change.*')`. I didn't think about using `selectAll('change')`. Nice.